### DISCUSSION TITLE
Setting control options

### DISCUSSION THREAD

**Discussion by andyslack** [NathanAP/vue-google-maps-community-fork#36] [Q&A]
Hi Guys,

Thanks so much for maintaining this folk.

A quick question on how to set more advanced options, e.g. `mapTypeControlOptions`

I cannot seem to access the actual map property

Here are the docs: 
https://developers.google.com/maps/documentation/javascript/controls#maps_control_positioning-typescript

Here is a related thread for react:
https://github.com/google-map-react/google-map-react/issues/6

Any guidance on how I can reposition mapTypeControl would be appreciated.

**Comment by NathanAP** [comment]
Hello! Sorry on the delay to answer ):

As we can see [in this part of the code](https://github.com/NathanAP/vue-google-maps-community-fork/blob/main/src/components/map.vue#L166), it seems that we use `options` to create the map options, just like we can see in [this example](https://developers.google.com/maps/documentation/javascript/examples/control-positioning). Did you try to use something like this inside the `options` prop?

**Comment by andyslack** [comment]
Hi @NathanAP 

You are correct we can pass data into the options object without issue, however as we don't have access to the `google.maps` object, there is no way to correctly reference the values of things like:

```
google.maps.MapTypeControlStyle.HORIZONTAL_BAR
google.maps.ControlPosition.TOP_CENTER
```

Passing the string equivalent does not work "HORIZONTAL_BAR" or "TOP_CENTER"

I am not sure the nest way to make these available but I know they had the same issue here: https://github.com/google-map-react/google-map-react/issues/6 and managed to fix it.

Hope this helps.

**Comment by NathanAP** [comment]
Ah, yes. I remember that this situation was related to a limitation. I think that you could add the Google Maps to your index.html, but I believe it would break something else ): I'm almost sure there was an issue in Fawmi's repo that had something like yours.

I'm not sure how we could fix this... Maybe our library could export the `google.maps` object, so we could use it anywhere?

**Reply by andyslack** [reply]
> I'm not sure how we could fix this... Maybe our library could export the `google.maps` object, so we could use it anywhere?

Yes, that was my first thought, if we could access that object, we could reference it correctly when passing it back.

**Reply by NathanAP** [reply]
I'll open an issue based on this discussion and lock the conversation here, so we can always keep it in one place.
